src/3p.js

```diff
@@ -109,17 +109,20 @@ function executeAfterWriteScript(win, fn) {
  * @param {string} src
  */
 export function validateSrcPrefix(prefix, src) {
-
   if (!isArray(prefix)) {
     prefix = [prefix];
   }
-
-  for (const p of prefix) {
-    if (src.indexOf(p) === 0) {
-      return;
+  const totalPrefixes = prefix.length;
+  if (src !== undefined) {
+    let p = 0;
+    for (p = 0; p <= totalPrefixes; p++) {
+      const protocolIndex = src.indexOf(prefix[p]);
+      if (protocolIndex == 0) {
+        return;
+      }
     }
+    throw new Error('Invalid src ' + src);
```

Comment:
I forgot to actually run the validate call, snippet update above. Does the above sample raise an error on the console? It will output undefined and not throw an error.